src/core/geometry/qgsgeometry.h

```diff
@@ -785,6 +785,15 @@ class CORE_EXPORT QgsGeometry
      */
     bool deleteVertex( int atVertex );
 
+    /**
+     * Converts the vertex at the given position from/to circular
+     * \returns FALSE if atVertex does not correspond to a valid vertex
+     * on this geometry (including if this geometry is a Point),
+     * or if the specified vertex can't be converted (e.g. start/end points).
+     * \since QGIS 3.20
+     */
+    bool convertVertex( int atVertex );
```

Comment:
This needs a better descriptive name, like `toggleCircularAtVertex`
Maybe better have two separate methods to convert from / to circular?

Reply:
Ok will rename.

I'd also rather not have two unused methods yet. This can easily be added afterwards if one day we need it.